kernel/libc/newlib/newlib_getentropy.c

```diff
@@ -10,6 +10,8 @@
 #include <unistd.h>
 #include <sys/time.h>
 
+#include <arch/arch.h>
```

Comment:
Needed for _arch_mem_top. It was somehow being pulled in before by one of the other includes.